muni_portal/tests/integration/test_webpush.py

```diff
@@ -0,0 +1,69 @@
+from django.contrib.auth.models import User
+from django.test import Client, TestCase
+from django.urls import reverse
+from django_q.models import Success as SuccessTask
+from faker import Faker
+from rest_framework import status
+from rest_framework.test import APIClient
+from unittest.mock import patch
+
+from muni_portal.core.models import WebPushNotification, WebPushSubscription
+
+faker = Faker()
+
+
+class WebPushNotificationIntegrationTestCase(TestCase):
+
+    def setUp(self):
+        self.dj_client = Client()
+        self.api_client = APIClient()
+        self.password = faker.password(length=8)
+        self.user = User.objects.create_superuser(
+            username="test", email="test@test.com", password=self.password
+        )
+
+    @patch("pywebpush.WebPusher")
+    def test_webpush_notification_flow(self, web_pusher_mock):
+        # Login existing user
+        data = {"login": self.user.username, "password": self.password}
+        response = self.api_client.post(reverse("rest_registration:login"), data)
+        self.assertEquals(response.status_code, status.HTTP_200_OK)
+
+        # Create subscription via API
+        jwt_token = response.data.get("token")
+        self.api_client.credentials(HTTP_AUTHORIZATION=f"Bearer {jwt_token}")
+        subscription_data = {
+            "endpoint": faker.uri(),
+            "auth": faker.pystr(min_chars=1, max_chars=100),
+            "p256dh": faker.pystr(min_chars=1, max_chars=100),
+            "expiration_time": faker.date(),
+        }
+        response = self.api_client.post(reverse("webpush"), subscription_data)
+        self.assertEquals(response.status_code, status.HTTP_201_CREATED)
+        self.assertEquals(WebPushSubscription.objects.count(), 1)
```

Comment:
please create the subscription via the ORM.

The purpose of this test is to test the integration of Django admin -> queue task